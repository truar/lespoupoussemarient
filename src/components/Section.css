.Section {
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    color: white;
}

.Section {
    text-shadow: 1px 2px 2px #1a1a1a; 
}