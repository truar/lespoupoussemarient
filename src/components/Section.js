import React from 'react';
import VisibilitySensor from 'react-visibility-sensor';
import ScrollableAnchor from 'react-scrollable-anchor';
import './Section.css';

function Section(props) {
    const { id, children, onVisibilityChange, ...attributes } = props;
    attributes.className = `${attributes.className} Section`;
    return (
        <ScrollableAnchor id={`view-${id}`}>
            <VisibilitySensor partialVisibility={true} onChange={onVisibilityChange}>
                <section id={id} {...attributes}>
                    {children}
                </section>
            </VisibilitySensor>
        </ScrollableAnchor>
    );
}

export default Section;