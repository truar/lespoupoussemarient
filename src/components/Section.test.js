import React from 'react';
import Section from './Section';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const component = shallow(<Section />);
  expect(component).toMatchSnapshot();
});
