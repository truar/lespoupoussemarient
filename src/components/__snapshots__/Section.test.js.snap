// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <VisibilitySensor
        active={true}
        containment={null}
        delayedCall={false}
        intervalCheck={true}
        intervalDelay={100}
        minTopValue={0}
        offset={Object {}}
        partialVisibility={true}
        resizeCheck={false}
        resizeDelay={250}
        resizeThrottle={-1}
        scrollCheck={false}
        scrollDelay={250}
        scrollThrottle={-1}
      >
        <section
          className="undefined Section"
        />
      </VisibilitySensor>,
      "id": "view-undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": <section
          className="undefined Section"
        />,
        "containment": null,
        "delayedCall": false,
        "intervalCheck": true,
        "intervalDelay": 100,
        "minTopValue": 0,
        "offset": Object {},
        "onChange": undefined,
        "partialVisibility": true,
        "resizeCheck": false,
        "resizeDelay": 250,
        "resizeThrottle": -1,
        "scrollCheck": false,
        "scrollDelay": 250,
        "scrollThrottle": -1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "undefined Section",
          "id": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <VisibilitySensor
          active={true}
          containment={null}
          delayedCall={false}
          intervalCheck={true}
          intervalDelay={100}
          minTopValue={0}
          offset={Object {}}
          partialVisibility={true}
          resizeCheck={false}
          resizeDelay={250}
          resizeThrottle={-1}
          scrollCheck={false}
          scrollDelay={250}
          scrollThrottle={-1}
        >
          <section
            className="undefined Section"
          />
        </VisibilitySensor>,
        "id": "view-undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": <section
            className="undefined Section"
          />,
          "containment": null,
          "delayedCall": false,
          "intervalCheck": true,
          "intervalDelay": 100,
          "minTopValue": 0,
          "offset": Object {},
          "onChange": undefined,
          "partialVisibility": true,
          "resizeCheck": false,
          "resizeDelay": 250,
          "resizeThrottle": -1,
          "scrollCheck": false,
          "scrollDelay": 250,
          "scrollThrottle": -1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "undefined Section",
            "id": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
